lib/discovery/index.js

```diff
@@ -889,7 +888,7 @@ export class PeerInfo extends TypedEmitter {
    * @param {Object} options
    * @param {IdentityPublicKeyString} [options.identityPublicKey]
    * @param {('dht'|'mdns')} [options.discoveryType]
-   * @param {ConnectionStream} [options.connection]
+   * @param {SecretStream<RawConnectionStream>} [options.connection]
```

Comment:
ideally would do some conditional typing to narrow down the RawConnectionStream type based on the `discoveryType`. not sure how to go about that in TS + JSDoc

Reply:
ah, yeah that would be nice. i think we're good for now with `SecretStream<RawConnectionStream>` in part because ideally at this stage the streams from the two discovery types should be pretty much equivalent.